tools/add_triage_signature.py

```diff
@@ -48,33 +53,60 @@
 """
 
 
+def config_logger(verbose: bool):
+    global logger
+
+    handler = colorlog.StreamHandler()
+    handler.setFormatter(colorlog.ColoredFormatter('%(log_color)s%(levelname)-10s %(message)s'))
+
+    logger.setLevel(logging.INFO)
+    logger.addHandler(handler)
+
+    if verbose:
+        logger.setLevel(logging.DEBUG)
+
+
 def format_description(failure_data):
     return JIRA_DESCRIPTION.format(**failure_data)
 
+
 def days_ago(datestr):
     try:
         return (datetime.now() - dateutil.parser.isoparse(datestr)).days
     except Exception:
         logger.debug("Cannot parse date: %s", datestr)
         return 9999
 
+
 @functools.lru_cache(maxsize=1000)
 def get_metadata_json(cluster_url):
     res = requests.get("{}/metdata.json".format(cluster_url))
     res.raise_for_status()
     return res.json()
 
+
 @functools.lru_cache(maxsize=1000)
 def get_events_json(cluster_url, cluster_id):
     res = requests.get(f"{cluster_url}/cluster_{cluster_id}_events.json")
     res.raise_for_status()
     return res.json()
 
+
+@functools.lru_cache(maxsize=1000)
```

Comment:
How is that different from the other "get files" functions?

Reply:
See my comment below. It's different because those functions actually return the values.

The `lru_cache` wrapper takes the arguments passed to the function as a `(arg1, arg2, ...)` tuple, then uses that as a key to a dictionary that contains recorded return values as values.